File: FrontEnd/Budgetkart/src/app/common/purchase.ts
import {Customer} from "./customer";
import {Order} from "./order";
import {OrderItem} from "./order-item";
import {Address} from "./address";

export class Purchase
{
    // @ts-ignore
    customer:Customer;
    // @ts-ignore
    shippingAddress:Address;
    // @ts-ignore
    billingAddress:Address;
    // @ts-ignore
    order:Order;
    // @ts-ignore
    orderItems:OrderItem[];
}

File: backend/BudgetKart/src/main/java/org/utkarsh2573/budgetkart/dto/Purchase.java
package org.utkarsh2573.budgetkart.dto;

import lombok.Data;
import org.utkarsh2573.budgetkart.entity.Address;
import org.utkarsh2573.budgetkart.entity.Customer;
import org.utkarsh2573.budgetkart.entity.Order;
import org.utkarsh2573.budgetkart.entity.OrderItem;

import java.util.Set;

@Data
public class Purchase
{
    private Customer customer;

    private Address shippingAddress;

    private Address billingAddress;

    private Order order;

    private Set<OrderItem> orderItems;
}
